core/src/main/java/com/opensymphony/xwork2/conversion/impl/DefaultTypeConverter.java

```diff
@@ -63,7 +63,7 @@
         map.put(Float.TYPE, new Float(0.0f));
         map.put(Double.TYPE, new Double(0.0));
         map.put(BigInteger.class, new BigInteger("0"));
-        map.put(BigDecimal.class, new BigDecimal(0.0));
+        map.put(BigDecimal.class, BigDecimal.valueOf(0.0));
```

Comment:
Tiny nitpick: there are `BigDecimal.ZERO` and `BigInteger.ZERO`.

Reply:
I considered it. But b/c of https://stackoverflow.com/questions/9992006/equals-operator-for-zeros-bigdecimal-double-in-java I was not quite sure if 0 is the right substitute for 0.0 at this place. What do you think - should we try anyways?

Reply:
[`equals` shouldn't be used to compare BigDecimals](http://www.opentaps.org/docs/index.php/How_to_Use_Java_BigDecimal:_A_Tutorial)

And it is seems you've already changed it :)
```
BigDecimal.ZERO.equals(BigDecimal.valueOf(0.0)) // false
BigDecimal.ZERO.equals(new BigDecimal(0.0)) // true
```

Reply:
@aleksandr-m Considering that no tests are failing, what do you prefer? Should I revert this line (no risk of changed behaviour in depending code) or maybe better switch to `BigDecimal.ZERO` + `BigInteger.ZERO` for L:66.

@lukaszlenart do you have any thoughts about that?

Reply:
I would assume `BigDecimal.valueOf(0.0)` is better here

Reply:
I think we can go for `BigInteger.ZERO` and `BigDecimal.ZERO as @aleksandr-m proposed.

see commit d09d3dc